src/Caching/Impl/ModuleFactory.cs

```diff
@@ -205,7 +205,7 @@ private IMember GetMemberFromModule(IPythonModule module, IReadOnlyList<string>
                 case "Unknown":
                     return builtins.Interpreter.UnknownType;
                 case "SuperType":
-                    return new PythonSuperType(typeArgs);
+                    return new PythonSuperType(typeArgs, builtins);
```

Comment:
Bit fishy, technically super should have proper declaring module...

Reply:
Why can only modules have an unset declaring module?

Reply:
Each type belong somewhere, i.e. where is the source, such as for navigation and accessing interpreter besides other things. Modules have null declaring module although technically submodules could have it or module could have itself there.